.github/workflows/quality.yml

```diff
@@ -66,12 +66,15 @@ jobs:
       - name: Set up rust cache
         uses: Swatinem/rust-cache@v2
         with:
-          cache-on-failure: true
+          save-if: ${{ github.ref == 'refs/heads/main' }}
```

Comment:
Would it make sense to also include the `release/*` branches?

Reply:
The goal is to only cache what is linked to the `main` lock file hash. The release branches aim at producing a docker image which doesn't use the `rust-cache` tho.
They can be some improvements here but I didn't pushed too far.